mesa/space.py

```diff
@@ -251,22 +251,33 @@ def get_neighborhood(
         if neighborhood is not None:
             return neighborhood
 
+        # Using a list for neighhborhood in respect to any other built-in data
+        # structure allows to optimize the code further (e.g. with Cython or Numba)
+        # more easily. To better understand how the algorithm was conceived, look at
+        # https://github.com/projectmesa/mesa/pull/1476#issuecomment-1306220403 and in
+        # general to the PR#1476.
         neighborhood = []
 
         x, y = pos
         if self.torus:
-            x_radius = min(radius, self.width // 2)
-            y_radius = min(radius, self.height // 2)
+            x_max_radius, y_max_radius = self.width // 2, self.height // 2
+            x_radius, y_radius = min(radius, x_max_radius), min(radius, y_max_radius)
 
-            for dx in range(-x_radius, x_radius + 1):
-                for dy in range(-y_radius, y_radius + 1):
+            # For each dimension, in the edge case where the radius is as big as
+            # possible and the dimension is even we need to shrink by one the range
+            # of values to avoid duplicates in neighborhood
```

Comment:
One last thing: needs to add the example `(0 % 4 and 4 % 4)`.

Reply:
Added